Libraries/LibWebView/WebUI/SettingsUI.cpp

```diff
@@ -263,4 +270,40 @@ void SettingsUI::set_do_not_track(JsonValue const& do_not_track)
     WebView::Application::settings().set_do_not_track(do_not_track.as_bool() ? DoNotTrack::Yes : DoNotTrack::No);
 }
 
+void SettingsUI::set_dns_settings(JsonValue const& dns_settings)
+{
+    // dnsSettings :: { mode: "system" } | { mode: "custom", server: string, port: u16, type: "udp" | "tls", forciblyEnabled: bool }
+    if (!dns_settings.is_object())
+        return;
+
+    auto& obj = dns_settings.as_object();
+    auto mode = obj.get_string("mode"sv);
+    if (!mode.has_value())
+        return;
+
+    if (*mode == "system") {
+        WebView::Application::settings().set_dns_settings(SystemDNS {});
+    } else if (*mode == "custom") {
+        auto server = obj.get_string("server"sv).map([](auto const& name) { return name.to_byte_string(); });
```

Comment:
Instead of writing 2 JSON parsers for each non-trivial object, I've been adding static parser methods to Settings.cpp. For example,`Settings::parse_json_languages` and `Settings::parse_custom_search_engine`. Seems like DNS settings would benefit from the same thing to return a `DNSSettings` type?